### PR TITLE
Task/29  profile has been created

### PR DESCRIPTION
**Pull request by jananikuruntham**

### CODE REVIEW

File: api/src/users/users.service.ts [hunk 1]
```diff
@@ -192,7 +193,67 @@ export class UsersService {
...
+      if (existingUser) {
+        const isPasswordMatch = await compare(
+          currentPassword,
+          existingUser.password,
+        );
+        if (!isPasswordMatch) {
+          return {
+            message: 'Current password is incorrect.',
+            statusCode: 422,
+            data: {},
```
Comment (vsaravanakumar) [inline api/src/users/users.service.ts:242]: Remove data object from the response if it's not needed

File: front/src/components/Profile/Profile.tsx [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+  role: "",
+  password: "",
+};
+
+interface AlertState {
+  severity: "success" | "info" | "warning" | "error";
+  message: string;
+}
+
+interface AddProfileModalProps {
```
Comment (vsaravanakumar) [inline front/src/components/Profile/Profile.tsx:59]: Remove the unused interface

File: front/src/components/Profile/Profile.tsx [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+  onEdit: (editedData: RowData) => void;
+}
+
+const bull = (
```
Comment (vsaravanakumar) [inline front/src/components/Profile/Profile.tsx:63]: Remove the unused variable

File: front/src/components/Profile/Profile.tsx [hunk 4]
```diff
@@ -0,0 +1,622 @@
...
+    email: "",
+    mobile: "",
+    user_name: "",
+    display_name: "",
+    role: "",
+    password: "",
+  });
+
+  const [errors, setErrors] = useState<Errors>({});
+  const [user_name, setUserName] = useState(userDetails.user_name);
```
Comment (vsaravanakumar) [inline front/src/components/Profile/Profile.tsx:90]: Remove the unused variables(user_name, role, email, mobile, rows and password)

File: front/src/components/Profile/Profile.tsx [hunk 5]
```diff
@@ -0,0 +1,622 @@
...
+            ...prevEditedData,
+            display_name,
+            user_name,
+            email,
+            mobile,
+            id,
+            role,
+            password,
+          }));
+          setUserName(user_name || "");
```
Comment (vsaravanakumar) [inline front/src/components/Profile/Profile.tsx:136]: Remove the unused variable data set

File: front/src/components/Profile/Profile.tsx [hunk 6]
```diff
@@ -0,0 +1,622 @@
...
+      setRole(userDetails.role || "");
+    }
+  }, [userDetails]);
+
+  useEffect(() => {
+    getApplications();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const getApplications = async () => {
```
Comment (vsaravanakumar) [inline front/src/components/Profile/Profile.tsx:166]: Give a relevant name for the function

File: front/src/components/Profile/Profile.tsx [hunk 7]
```diff
@@ -0,0 +1,622 @@
...
+      newErrors.Email = "Email is required";
+    } else if (!validateEmail(editedData.email)) {
+      newErrors.Email = "Invalid email address";
+    }
+    setErrors(newErrors);
+
+    return Object.keys(newErrors).length === 0;
+  };
+
+  const editUser = async (id: number, updatedData: RowData) => {
```
Comment (vsaravanakumar) [inline front/src/components/Profile/Profile.tsx:233]: Give a relevant name for the function

File: front/src/components/Profile/Profile.tsx [hunk 8]
```diff
@@ -0,0 +1,622 @@
...
+    }
+    for (let i = 0; i < newPassword.length; i++) {
+      if (newPassword[i] !== confirmPassword[i]) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  const editPassword = async (
```
Comment (vsaravanakumar) [inline front/src/components/Profile/Profile.tsx:303]: Give a relevant name for the function

File: front/src/components/Profile/example.tsx [hunk 9]
```diff
@@ -0,0 +1,188 @@
+
```
Comment (vsaravanakumar) [inline front/src/components/Profile/example.tsx:1]: Remove the unwanted file

File: front/src/components/Profile/index.tsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import Link from "next/link";
```
Comment (vsaravanakumar) [inline front/src/components/Profile/index.tsx:1]: Remove the unwanted file

File: api/src/users/users.service.ts (hunk 1 above)
Comment (jananikuruntham) [reply api/src/users/users.service.ts]: removed data object

File: front/src/components/Profile/Profile.tsx (hunk 2 above)
Comment (jananikuruntham) [reply front/src/components/Profile/Profile.tsx]: removed interface

File: front/src/components/Profile/Profile.tsx (hunk 3 above)
Comment (jananikuruntham) [reply front/src/components/Profile/Profile.tsx:59]: removed all unused variables

File: front/src/components/Profile/Profile.tsx (hunk 4 above)
Comment (jananikuruntham) [reply front/src/components/Profile/Profile.tsx]: removed unused variables

File: front/src/components/Profile/Profile.tsx (hunk 5 above)
Comment (jananikuruntham) [reply front/src/components/Profile/Profile.tsx]: removed unused variables in data set

File: front/src/components/Profile/Profile.tsx (hunk 6 above)
Comment (jananikuruntham) [reply front/src/components/Profile/Profile.tsx]: relevant name for function has been changed

File: front/src/components/Profile/Profile.tsx (hunk 7 above)
Comment (jananikuruntham) [reply front/src/components/Profile/Profile.tsx]: relevant name for function has been changed

File: front/src/components/Profile/Profile.tsx (hunk 8 above)
Comment (jananikuruntham) [reply front/src/components/Profile/Profile.tsx]: relevant name for function has been changed

File: front/src/components/Profile/example.tsx (hunk 9 above)
Comment (jananikuruntham) [reply front/src/components/Profile/example.tsx:1]: removed unwanted file

File: front/src/components/Profile/index.tsx (hunk 10 above)
Comment (jananikuruntham) [reply front/src/components/Profile/index.tsx:1]: removed unwanted file

**Pull request merged**